pkgs/applications/office/paperwork/paperwork-backend.nix

```diff
@@ -30,48 +31,56 @@ buildPythonPackage rec {
 
   sourceRoot = "source/paperwork-backend";
 
-  # Python 2.x is not supported.
-  disabled = !isPy3k && !isPyPy;
-
   patchPhase = ''
     echo 'version = "${version}"' > src/paperwork_backend/_version.py
     chmod a+w -R ..
     patchShebangs ../tools
   '';
 
   propagatedBuildInputs = [
+    distro
+    gtk3
+    libinsane
+    natsort
+    openpaperwork-core
     pyenchant
-    scikit-learn
-    pypillowfight
     pycountry
-    whoosh
-    termcolor
-    python-Levenshtein
-    libinsane
     pygobject3
     pyocr
-    natsort
-    pkgs.poppler_gi
-    pkgs.gtk3
-    distro
-    openpaperwork-core
+    pypillowfight
+    python-Levenshtein
+    poppler_gi
+    scikit-learn
+    termcolor
+    whoosh
   ];
 
-  preCheck = ''
-    export HOME=$(mktemp -d)
-  '';
+  nativeBuildInputs = [
+    gettext
+    shared-mime-info
+    which
+  ];
+
+  checkInputs = [
+    libreoffice
+    openpaperwork-gtk
+    psutil
+    python
+  ];
 
-  nativeBuildInputs = [ pkgs.gettext pkgs.which pkgs.shared-mime-info ];
   preBuild = ''
     make l10n_compile
   '';
 
-  checkInputs = [ openpaperwork-gtk psutil pkgs.libreoffice ];
+  checkPhase = ''
+    export HOME=$(mktemp -d)
+    ${python.interpreter} -m unittest
```

Comment:
please use unittestCheckHook instead. (it's a recent addition, you need to rebase your PR)